cephfs/admin/fs_quiesce.go

```diff
@@ -0,0 +1,131 @@
+//go:build ceph_preview
+
+package admin
+
+import "fmt"
+
+// FixedPointFloat is a custom type that implements the MarshalJSON interface.
+// This is used to format float64 values to two decimal places.
+// By default these get converted to integers in the JSON output and
+// fail the command.
+type FixedPointFloat float64
+
+// MarshalJSON provides a custom implementation for the JSON marshalling
+// of FixedPointFloat. It formats the float to two decimal places.
+func (fpf FixedPointFloat) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("%.2f", float64(fpf))), nil
+}
+
+// SubVolumeQuiesceFields is the internal type used to create JSON for ceph.
+// See SubVolumeQuiesceOptions for the type that users of the library
+// interact with.
+type SubVolumeQuiesceFields struct {
+	Prefix     string          `json:"prefix"`
+	VolName    string          `json:"vol_name"`
+	GroupName  string          `json:"group_name,omitempty"`
+	Members    []string        `json:"members,omitempty"`
+	SetId      string          `json:"set_id,omitempty"`
+	Timeout    FixedPointFloat `json:"timeout,omitempty"`
+	Expiration FixedPointFloat `json:"expiration,omitempty"`
+	AwaitFor   FixedPointFloat `json:"await_for,omitempty"`
+	Await      bool            `json:"await,omitempty"`
+	IfVersion  int             `json:"if_version,omitempty"`
+	Include    bool            `json:"include,omitempty"`
+	Exclude    bool            `json:"exclude,omitempty"`
+	Reset      bool            `json:"reset,omitempty"`
+	Release    bool            `json:"release,omitempty"`
+	Query      bool            `json:"query,omitempty"`
+	All        bool            `json:"all,omitempty"`
+	Cancel     bool            `json:"cancel,omitempty"`
+}
+
+// SubVolumeQuiesceOptions are used to specify optional, non-identifying, values
+// to be used when quiescing a subvolume.
+type SubVolumeQuiesceOptions struct {
+	Timeout    FixedPointFloat
```

Comment:
if you can use `float64` here, there would not be a need to export the custom `FixedPointFloat` type, and it can stay internal. That would be much cleaner.

Reply:
Like I mentioned [here](https://github.com/ceph/go-ceph/pull/958#discussion_r1498553362), float64 unmarshall works only if we actually pass float number, it does not unmarshall to integer like 15 will be 15 and not 15.0, which ceph manager will reject as invalid. Example:
`
cmd=[{"prefix":"fs quiesce","vol_name":"cephfs","members":["quiesceMe"],"timeout":10.7,"expiration":15}]: dispatch
`

Reply:
https://stackoverflow.com/questions/20099328/does-go-correctly-marshal-float64-to-json

Reply:
Yes, I understand that part. But my question is about the requirement of a public/exported type `FixedPointFloat`. Can you call it `fixedPointFloat` and does it still work?

Reply:
Is it okay that the type is not exported but the structure SubVolumeQuiesceOptions is exported of which exported fields are of type fixedPointFloat?

Reply:
I made external struct SubVolumeQuiesceOptions use float64 and internal fsQuiesceFields use fixedPointFloat, so I don't have to export fixedPointFloat